packages/video_player/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java

```diff
@@ -49,15 +52,18 @@
     }
 
     VideoPlayer(
+        Context context,
         EventChannel eventChannel,
         TextureRegistry.SurfaceTextureEntry textureEntry,
         String dataSource,
-        Result result) {
+        Result result,
+        Map<String, String> headers) {
       this.eventChannel = eventChannel;
       this.mediaPlayer = new MediaPlayer();
       this.textureEntry = textureEntry;
+      this.headers = headers;
       try {
-        mediaPlayer.setDataSource(dataSource);
+        mediaPlayer.setDataSource(context, Uri.parse(dataSource), headers);
```

Comment:
network video files might fail to load using this method:
https://stackoverflow.com/a/42160256/1068346